Astro CLI v1.0.0 (All Changes) (#690)
astronomer/docs: software/deploy-cli.md

sidebar_label: 'Deploy DAGs via CLI'
title: 'Deploy DAGs to Astronomer Software via CLI'
id: deploy-cli
description: How to push DAGs to your Airflow Deployment on Astronomer Software using the Astro CLI.
---

## Overview

This guide provides the setup steps for deploying DAGs to Astronomer using the Astro CLI.

> If you've used the Astro CLI to develop locally, you'll find it similarly easy to deploy your DAGs to an Airflow Deployment on Astronomer. The Astro CLI builds your DAGs into a Docker image alongside all other files in your Airflow project directory, including your Python and OS-level packages, your Dockerfile, and your plugins. The resulting image is then used to generate a set of Docker containers for each of Airflow's core components.

Reviewer: ```suggestion
If you've used the Astro CLI to develop locally, the process for deploying your DAGs to an Airflow Deployment on Astronomer should be equally familiar. The Astro CLI builds your DAGs into a Docker image alongside all the other files in your Airflow project directory, including your Python and OS-level packages, your Dockerfile, and your plugins. The resulting image is then used to generate a set of Docker containers for each core Airflow component.
```

For guidance on automating this process, refer to [Deploy to Astronomer via CI/CD](ci-cd.md). To learn how to add Python and OS-level packages or otherwise customize your Docker image, read [Customize your Image](customize-image.md).

Alternatively, you can configure an external NFS volume for DAG deploys. For more information, read [Deploy DAGs to an NFS Volume](deploy-nfs.md).

> > **Note:** We recommend that all users test their code locally via the Astro CLI before pushing it to an Airflow Deployment on Astronomer. For guidelines on developing locally, refer to [CLI Quickstart](cli-quickstart.md).

Reviewer: ```suggestion
> **Note:** Astronomer recommends that all users use the Astro CLI to test their code locally before pushing it to an Airflow Deployment on Astronomer. For guidelines on developing locally, see [CLI Quickstart](cli-quickstart.md).
```

## Prerequisites

In order to push up DAGs to a Deployment on Astronomer, you must have:

* [The Astro CLI](cli-quickstart.md) installed.
* Access to an Astronomer platform at `https://app.BASEDOMAIN`.
* An Astronomer [Workspace](manage-workspaces.md) with at least one active [Airflow Deployment](configure-deployment.md).

## Step 1: Authenticate to Astronomer

> To authenticate via the Astro CLI, run:

Reviewer: ```suggestion
To authenticate with the Astro CLI, run:
```

```sh
astro login BASEDOMAIN
```

## Step 2: Confirm Your Workspace and Deployment

> From the Astro CLI, you're free to push code to any Airflow Deployment you have access to as long as you have the appropriate deployment-level permissions to do so. For more information on both Workspace and Deployment-level permissions on Astronomer, refer to [User Permissions](workspace-permissions.md).

Reviewer: ```suggestion
From the Astro CLI, you can push code to any Airflow Deployment you have access to as long as you have the appropriate deployment-level permissions. For more information on both Workspace and Deployment-level permissions on Astronomer, see [User Permissions](workspace-permissions.md).
```
